go/vt/vtadmin/api.go

```diff
@@ -1050,7 +1065,53 @@ func (api *API) GetSchemaMigrations(ctx context.Context, req *vtadminpb.GetSchem
 	span, ctx := trace.NewSpan(ctx, "API.GetSchemaMigrations")
 	defer span.Finish()
 
-	panic("implement me!")
+	clusterIDs := make([]string, 0, len(req.ClusterRequests))
+	requestsByCluster := make(map[string][]*vtctldatapb.GetSchemaMigrationsRequest, len(req.ClusterRequests))
+
+	for _, r := range req.ClusterRequests {
+		clusterIDs = append(clusterIDs, r.ClusterId)
+		requestsByCluster[r.ClusterId] = append(requestsByCluster[r.ClusterId], r.Request)
+	}
+
+	clusters, _ := api.getClustersForRequest(clusterIDs)
+
+	var (
+		m       sync.Mutex
+		wg      sync.WaitGroup
```

Comment:
Any reason not to use an `errgroup`? Seems like it would be a perfect fit -- making the code simpler to read and ending earlier.

Reply:
the main difference between `errgroup` and this pattern is the following (emphasis mine):

> Wait blocks until all function calls from the Go method have returned, then returns the **first non-nil error (if any)** from them.

we would be losing information by switching, as we currently collect all errors. and (as i understand it) `errgroup` doesn't actually terminate other goroutines early if one returns an error, it just limits the amount of error information you get, so it doesn't end up ending earlier

Reply:
Yeah, it doesn't forcefully end anything AFAIUI — it cancels the context (assuming errgroup.WithContext() was used) so it should cause things to end earlier _assuming_ that the errgroup's context is used by calls within the errgroup's Go function AND that the code executed in the goroutines check for context cancellation. Even so, the point about losing error information is valid. I was just curious -- the current code is fine.